internal/command/views/test.go

```diff
@@ -570,3 +579,45 @@ func testStatus(status moduletest.Status) string {
 		panic("unrecognized status: " + status.String())
 	}
 }
+
+// SaveErroredTestStateFile is a helper function to invoked in DestorySummary
+// to store the state to errored_test.tfstate and handle associated diagnostics and errors with this operation
+func SaveErroredTestStateFile(state *states.State, op Operation) tfdiags.Diagnostics {
+	var diags tfdiags.Diagnostics
+	localFileSystem := statemgr.NewFilesystem("errored_test.tfstate")
+	stateFile := statemgr.NewStateFile()
+	stateFile.State = state
+	writeErr := localFileSystem.WriteStateForMigration(stateFile, true)
+	if writeErr != nil {
+		if dumpErr := op.EmergencyDumpState(stateFile); dumpErr != nil {
```

Comment:
It may be nice to have some comments here explaining why we dump the state here like this so nobody comes by and cleans this up in the future.

Reply:
Understood, will add a few comments explaining what is being done.